ts/data/tooltip.ts

```diff
@@ -0,0 +1,92 @@
+class Tooltip {
+  private mouseOnIcon: boolean;
+  private mouseOnToolTip: boolean;
+  private tooltipContainer: HTMLElement;
+
+  constructor(target: HTMLElement, contents: Node) {
+    this.mouseOnIcon = false;
+    this.mouseOnToolTip = false;
+    this.tooltipContainer = this.getTootipContainer();
+
+    target.onmouseenter = () => {
+      this.mouseOnIcon = true;
+      const targetY = this.getOffset(target).bottom;
+      const targetX = this.getOffset(target).left;
+      // empties tooltip container
+      this.tooltipContainer.replaceChildren();
+      // node needs to be cloned since appendChild removes it
+      this.tooltipContainer.appendChild(contents.cloneNode(true));
```

Comment:
Can add the contents in `replaceChildren` instead of using `appendChild` afterwards